### PR TITLE
bugfix/MET-4220_Magick_uses_all_disk_space

### PR DESCRIPTION
**Pull request by marcin-rp**
…iles left by Image Magick.

### CODE REVIEW

File: metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/CommandExecutor.java [hunk 1]
```diff
@@ -20,6 +21,7 @@
 class CommandExecutor {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(CommandExecutor.class);
+  public static final int PROCESS_TERMINATION_TIMEOUT_SECONDS = 30;
```
Comment (jeortizquan) [inline metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/CommandExecutor.java:24]: ```private``` perhaps? I see at this moment is not used in other place.

File: metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/ThumbnailGenerator.java [hunk 2]
```diff
@@ -310,6 +314,40 @@ private ImageMetadata generateThumbnailsInternal(List<ThumbnailWithSize> thumbna
     return result;
   }
 
+  private String executeImageMagick(List<String> command) throws MediaExtractionException {
+    Path tempDir = createMagickTempDirectory();
+    try {
+      return commandExecutor.execute(command, singletonMap(MAGICK_TEMPORARY_PATH, tempDir.toString()), false,
+              message -> new MediaExtractionException("Could not analyze content and generate thumbnails: " + message));
+    } finally {
```
Comment (jeortizquan) [inline metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/ThumbnailGenerator.java:322]: :+1:

File: metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/ThumbnailGenerator.java [hunk 3]
```diff
@@ -310,6 +314,40 @@ private ImageMetadata generateThumbnailsInternal(List<ThumbnailWithSize> thumbna
     return result;
   }
 
+  private String executeImageMagick(List<String> command) throws MediaExtractionException {
+    Path tempDir = createMagickTempDirectory();
+    try {
+      return commandExecutor.execute(command, singletonMap(MAGICK_TEMPORARY_PATH, tempDir.toString()), false,
```
Comment (jeortizquan) [inline metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/ThumbnailGenerator.java:320]: Note to investigate: Just to be sure. Is this temporary directory being created in ```/var/tmp/...``` or ```/tmp/...```? or do we need to checkout the property ```java.io.tmpdir``` of the system?

**Review by jeortizquan — COMMENTED** [review]
I think is good. I have some questions that need to confirm via the team.

File: metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/ThumbnailGenerator.java (hunk 3 above)
Comment (jeortizquan) [reply metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/ThumbnailGenerator.java:320]: Also check sonar security HOTSPOT, and remove code smells.

Comment (marcin-rp) [reply metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/ThumbnailGenerator.java:320]: > Note to investigate: Just to be sure. Is this temporary directory being created in `/var/tmp/...` or `/tmp/...`? or do we need to checkout the property `java.io.tmpdir` of the system?

java.io.tmpdir is set to special temp directory of storm worker

**Comment by stzanakis** [comment]
Hi both!
Please be careful with those changes.
If you are creating temp files/directories please use TempFileUtils from the metis-common-utils which is handling secure temp files, thanks!

**Review by jeortizquan — APPROVED** [review]

**Pull request merged**